test/byContract/IexecPocoBoost/IexecEscrow.v8.test.ts

```diff
@@ -95,28 +87,27 @@ describe('IexecEscrow.v8', function () {
         });
 
         it('Should not unlock funds when insufficient balance', async function () {
-            await expect(
-                iexecEscrow.unlock_(user.address, initialUserFrozen + 1),
-            ).to.be.revertedWith('IexecEscrow: Transfer amount exceeds balance');
+            await expect(iexecEscrow.unlock_(user.address, amount)).to.be.revertedWith(
+                'IexecEscrow: Transfer amount exceeds balance',
+            );
         });
     });
 
     describe('Reward', function () {
         it('Should reward', async function () {
-            // Check balances before the operation.
-            await checkInitialBalancesAndFrozens();
-            // Run operation.
+            // Fund iexecEscrow so it can reward the user.
+            await iexecEscrow.setBalance(iexecEscrow.address, amount).then((tx) => tx.wait());
+
             await expect(iexecEscrow.reward_(user.address, amount, ref))
+                .to.changeTokenBalances(
+                    iexecEscrow,
+                    [iexecEscrow.address, user.address],
+                    [-amount, amount],
+                )
                 .to.emit(iexecEscrow, 'Transfer')
                 .withArgs(iexecEscrow.address, user.address, amount)
                 .to.emit(iexecEscrow, 'Reward')
                 .withArgs(user.address, amount, ref);
```

Comment:
No need for frozen check?

Reply:
Nope
https://github.com/iExecBlockchainComputing/PoCo/blob/995ca3498c6699235fa7a6148e92a74d0e42586b/contracts/modules/delegates/IexecEscrow.v8.sol#L46-L49